### PR TITLE
HDDS-11764. Fix initiating DocumentBuilderFactory performance problem.

### PR DESCRIPTION
**Pull request by juncevich**
HDDS-11764. Fix initiating DocumentBuilderFactory performance problem.

Cache initiating DocumentBuilderFactory and NodeSchemaManager to ip

[Link to Jira task: HDDS-11764](https://issues.apache.org/jira/browse/HDDS-11764)

Patch tested only by performance tests.

### CODE REVIEW

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NodeSchemaManager.java [hunk 1]
```diff
@@ -55,6 +58,18 @@ public static NodeSchemaManager getInstance() {
...
+  public static NodeSchemaManager getInitiatedInstance(String schemaFile) {
+    NodeSchemaManager cachedInstance = cache.get(schemaFile);
+    if (cachedInstance == null) {
+      instance = new NodeSchemaManager();
+      instance.init(schemaFile);
+      cache.put(schemaFile, instance);
+      return instance;
+    } else {
+      return cachedInstance;
+    }
```
Comment (adoroszlai) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NodeSchemaManager.java:70]: This is not thread-safe.  Use `cache.computeIfAbsent()` if `NodeSchemaManager.init()` has no side-effects.

**Review by adoroszlai — COMMENTED** [review]
Thanks @juncevich for working on this.  Please check CI run for failures: https://github.com/juncevich/ozone/actions/runs/11940547308

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NodeSchemaManager.java (hunk 1 above)
Comment (juncevich) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NodeSchemaManager.java:70]: Thanks, will fix.

**Comment by adoroszlai** [comment]
Looking at stack trace in the screenshot attached to HDDS-11764, I think this is fixed by #6434.  `OzoneManager.getClusterMap()` no longer creates new `NetworkTopologyManager`.

**Comment by juncevich** [comment]
> #6434

Yep. It looks like you're right.

**Comment by juncevich** [comment]
Will test and this PR close if the problem will solved.

**Comment by juncevich** [comment]
We found a performance problem with the recording file's speed. Without this fix, it degraded by approximately 14 %. We need time to find the problem.

**Pull request closed without merging**